blocks/feeds-source-content-api-by-day-block/sources/feeds-content-api-by-day.js

```diff
@@ -79,28 +75,68 @@ const resolve = function resolve(key) {
     try {
       body.query.bool.must_not = JSON.parse(excludeTerms)
     } catch (error) {
-      console.warn(`Failed to parse Exclude-Terms: ${key['Exclude-Terms']}`)
+      const err = new Error('Invalid Exclude-Terms')
+      err.statusCode = 500
+      throw err
     }
   }
 
-  // Append date field to basic query
-  const { dateField, dateRange } = key
-  if (dateField && dateRange) {
-    let rangeStart, rangeEnd
-    if (dateRange === 'latest') {
-      rangeEnd = 'now'
-      rangeStart = moment.utc().subtract(1, 'days').format('YYYY-MM-DD')
-    } else {
-      rangeEnd = rangeStart = dateRange
+  // Validate and append date field to basic query
+  let { dateField, dateRange } = key
+  if (!dateField || !dateRange) {
+    const err = new Error('Date is required')
+    err.statusCode = 500
+    throw err
+  }
+  if (validANSDates.indexOf(dateField) < 0) {
```

Comment:
I think equating it to -1 would be more clear. But that is just a suggestion, this would not break anything

Reply:
Changed to === -1